### PR TITLE
Use Pharos::YamlFile in phases/base parse_resource_file

### PR DESCRIPTION
**Pull request by kke**

### CODE REVIEW

File: lib/pharos/phases/base.rb [hunk 1]
```diff
@@ -21,8 +21,7 @@ def exec_script(script, vars = {})
 
       def parse_resource_file(path, vars = {})
```
Comment (kke) [inline lib/pharos/phases/base.rb]: This looks a whole lot like the one in `Pharos::Kube`, which I think is in a wrong place too

Comment (kke) [reply lib/pharos/phases/base.rb]: Hmm, it's a class method. This could just call it with `Pharos::Kube.parse_resource_file` I think?

**Comment by SpComb** [comment]
It's used for non-YAML files too.

File: lib/pharos/phases/base.rb (hunk 1 above)
Comment (SpComb) [reply lib/pharos/phases/base.rb]: That returns a `Kubeclient::Resource`, which is the wrong thing here.

File: lib/pharos/phases/base.rb [hunk 2]
```diff
@@ -21,8 +21,7 @@ def exec_script(script, vars = {})
...
         path = File.realpath(File.join(__dir__, '..', 'resources', path))
-        yaml = File.read(path)
-        Pharos::Erb.new(yaml).render(vars)
+        Pharos::YamlFile.new(path).load(vars)
```
Comment (SpComb) [inline lib/pharos/phases/base.rb]: This needs to return a string with the encoded YAML?

**Comment by kke** [comment]
So does it need to work with non-yamls?

**Comment by SpComb** [comment]
Fixes part of #203 but not completely:

```
    Pushing audit configs to master
    $ sudo mkdir -p /etc/pharos/audit
    ! 0
No such file or directory @ realpath_rec - /app/lib/pharos/resources/audit/webhook-config.yml
/app/lib/pharos/phases/base.rb:23:in `realpath'
/app/lib/pharos/phases/base.rb:23:in `parse_resource_file'
/app/lib/pharos/phases/configure_master.rb:74:in `install'
/app/lib/pharos/phases/configure_master.rb:29:in `call'
/app/lib/pharos/up_command.rb:132:in `handle_masters'
/app/lib/pharos/up_command.rb:46:in `block in configure'
/app/lib/pharos/up_command.rb:45:in `chdir'
/app/lib/pharos/up_command.rb:45:in `configure'
/app/lib/pharos/up_command.rb:25:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:132:in `run'
/app/lib/pharos/root_command.rb:14:in `run'
bin/pharos-cluster:12:in `<main>'
```

**Comment by SpComb** [comment]
> So does it need to work with non-yamls?

It's only used for .yml right now, but there's a need for templating non-yml files like systemd units as well.

**Comment by SpComb** [comment]
Tested to fix #203 now

**Comment by kke** [comment]
I guess we need to bring `Pharos::Erb` back in the future and make `YamlFIle` use that for the parse part (and move the `YamlFile::Namespace` under it)

**Comment by SpComb** [comment]
> I guess we need to bring Pharos::Erb back in the future and make YamlFIle use that for the parse part (and move the YamlFile::Namespace under it)

Yeah, most of the `Pharos::YamlFile` logic is not actually YAML-specific at all.

**Comment by kke** [comment]
`Pharos::ResourceFile` sounds good too. It could also have `.to_kube`.

File: lib/pharos/phases/base.rb [hunk 3]
```diff
@@ -21,8 +21,7 @@ def exec_script(script, vars = {})
 
       def parse_resource_file(path, vars = {})
         path = File.realpath(File.join(__dir__, '..', 'resources', path))
-        yaml = File.read(path)
-        Pharos::Erb.new(yaml).render(vars)
+        Pharos::YamlFile.new(path).read(vars)
```
Comment (SpComb) [inline lib/pharos/phases/base.rb]: The `Pharos::YamlFile#read` is entirely generic, and not actually YAML-specific, so this works nicely for any "resource" files...

Should probably be renamed to something like `upload_config_file`, using `lib/pharos/configs` or such... the `resources` are for Kube resources.

Comment (jakolehm) [reply lib/pharos/phases/base.rb]: Maybe open an issue about this?

Comment (SpComb) [reply lib/pharos/phases/base.rb]: #204

**Pull request merged**